### PR TITLE
Add logging to debug failed npm publish for legacy package names in CI

### PR DESCRIPTION
**Pull request by TooTallNate**
As you can see in https://github.com/zeit/now/runs/647945621,
the final step for publishing the legacy `@now` packages failed
with a 401 error from npm.

This additional logging an to attempt to debug why that is happening.

### CODE REVIEW

**Review by styfle — APPROVED** [review]

**Comment by styfle** [comment]
Interesting, the `@vercel` scoped packages were published properly using npm registry but the `@now` scoped packages failed because they were trying to publish to the yarn registry url.

```
lerna http fetch PUT 200 https://registry.npmjs.org/@vercel%2fbuild-utils 8938ms
```
```
npm ERR! 401 Unauthorized - PUT https://registry.yarnpkg.com/@now%2fbuild-utils - You must be logged in to publish packages.
```

The `lerna.json` references npm registry url here: https://github.com/zeit/now/blob/master/lerna.json#L9

File: utils/publish-legacy.sh [hunk 1]
```diff
@@ -3,9 +3,12 @@ set -euo pipefail
 
 # Modifies the tagged packages to contain the legacy `now` name.
 # This file will be deleted on Jan 1, 2021.
+echo "Publishing legacy \"@now\" packages"
 
 __dirname="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
```
Comment (styfle) [inline utils/publish-legacy.sh:8]: Let's also add a `echo __dirname` here and then also another echo on line 18 after `cd` so we can double check the directories.

Comment (TooTallNate) [reply utils/publish-legacy.sh:8]: https://github.com/zeit/now/pull/4266/commits/12090571e494eb1ca9e3716620c17df3d81650ef

File: utils/publish-legacy.sh [hunk 2]
```diff
@@ -19,6 +23,6 @@ for tag in $tags; do
     npm_tag="--tag canary"
   fi
 
-  echo "Running \`npm publish $npm_tag\` in \"$(pwd)\""
-  npm publish $npm_tag
+  echo "Running \`npm publish --registry=https://registry.npmjs.com $npm_tag\` in \"$(pwd)\""
+  echo "DRY: npm publish --registry=https://registry.npmjs.com $npm_tag"
```
Comment (TooTallNate) [inline utils/publish-legacy.sh:27]: Using `--registry` here is the _real_ fix. However, I've commented it out for now until we support `@vercel/static` and `@vercel/build-utils` in production.

**Pull request merged**